controllers/lock_client.go

```diff
@@ -156,6 +169,72 @@ func (client *RealLockClient) updateLock(transaction fdb.Transaction, start int6
 	transaction.Set(lockKey, lockValue.Pack())
 }
 
+// AddPendingUpgrades registers information about which process groups are
+// pending an upgrade to a new version.
+func (client *RealLockClient) AddPendingUpgrades(version fdbtypes.FdbVersion, processGroupIDs []string) error {
+	_, err := client.database.Transact(func(tr fdb.Transaction) (interface{}, error) {
+		for _, processGroupID := range processGroupIDs {
+			err := tr.Options().SetAccessSystemKeys()
+			if err != nil {
+				return nil, err
+			}
+
+			key := fdb.Key(fmt.Sprintf("%s/upgrades/%s/%s", client.cluster.GetLockPrefix(), version.String(), processGroupID))
+			tr.Set(key, []byte(processGroupID))
```

Comment:
Yes, this is intentional. Storing them all in one key would lead to more merge conflicts, and having the value makes it simple to extract the process group IDs, since we don't have to parse it out of the key.